src/Data/List/Relation/Binary/Equality/Setoid/Properties.agda

```diff
@@ -0,0 +1,46 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- Properties of List modulo ≋
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Relation.Binary.Bundles using (Setoid)
+
+module Data.List.Relation.Binary.Equality.Setoid.Properties
+  {c ℓ} (S : Setoid c ℓ)
+  where
+
+open import Algebra.Bundles using (Monoid)
+open import Algebra.Structures using (IsMonoid)
+open import Data.List.Base using (List; []; _++_)
+import Data.List.Properties as List
+import Data.List.Relation.Binary.Equality.Setoid as ≋
+open import Data.Product.Base using (_,_)
+open import Function.Base using (_∘_)
+open import Level using (_⊔_)
+
+open ≋ S using (_≋_; ≋-refl; ≋-reflexive; ≋-isEquivalence; ++⁺)
+
+------------------------------------------------------------------------
+-- The []-++-Monoid
+
+-- Structure
+
+isMonoid : IsMonoid _≋_ _++_ []
+isMonoid = record
+  { isSemigroup = record
+    { isMagma = record
```

Comment:
Could we have these `isMagma` and `isSemigroup` defined at the top level? In case someone needs them it'll be more convenient than digging into this. And also, I think it looks neater

Reply:
Makes sense, and while I was working on this PR, the ideas behind #2391 were coming into focus for me: so to make this sharper, do I 
* `open` the `Monoid monoid` structure to `public`ly export (a/the relevant subcollection of) its fields, 
* or build up to `isMonoid` by progressive instantiation?
Current library style is the first option, so I guess I'll follow that, but already I can imagine a time downstream where I/we opt for the second.